src/Workspaces/CSharp/Portable/Extensions/SyntaxTokenExtensions.cs

```diff
@@ -44,6 +44,11 @@ public static bool IsKind(this SyntaxToken token, params SyntaxKind[] kinds)
             return kinds.Contains(token.Kind());
         }
 
+        public static bool IsParentKind(this SyntaxToken token, SyntaxKind kind)
+        {
+            return token.Parent != null && token.Parent.IsKind(kind);
+        }
```

Comment:
!= null check is not necessary.  I'm skeptical this extension provides enough value given that it's always safe and clear to call token.Parent.IsKind(...)